src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/lsf/rest/LSFRestJobManager.groovy

```diff
@@ -181,34 +184,38 @@ class LSFRestJobManager extends BatchEuphoriaJobManagerAdapter {
      "--bqJky99mlBWa-ZuqjC53mG6EzbmlxB--\r\n"
      */
     private void submitJob(BEJob job) {
-        String headBoundary = UUID.randomUUID().toString()
-
         List<Header> headers = []
-        headers.add(new BasicHeader(HTTP.CONTENT_TYPE, "multipart/mixed;boundary=${headBoundary}"))
         headers.add(new BasicHeader("Accept", "text/xml,application/xml;"))
 
-        StringBuilder requestBody = new StringBuilder()
-        requestBody << getAppNameArea("generic", headBoundary)
+        List<String> requestParts = []
+        requestParts.add(createRequestPart("AppName", "generic"))
 
         // --- Parameters Area ---
-        StringBuilder paramArea = new StringBuilder()
-        String bodyBoundary = UUID.randomUUID().toString()
-
-        paramArea << prepareToolScript(job, bodyBoundary)
-        paramArea << prepareJobName(job, bodyBoundary)
-        paramArea << prepareExtraParams(job, bodyBoundary)
-
-        if (paramArea.length() > 0) {
-            paramArea.insert(0, getParamAreaHeader(headBoundary, bodyBoundary)) //header
-            paramArea << "--${bodyBoundary}--\r\n\r\n" //footer
+        List<String> jobParts = []
+        if (job.tool) {
+            jobParts.add(createJobPart("COMMAND", job.tool?.absolutePath as String, "COMMANDTORUN"))
+        } else {
```

Comment:
Why did you replace the well formatted structure which was existing? Now you have a lot of ifs and String constants instead of methods with speaking names.

Did you work with the latest version of the code?